src/StreamJsonRpc.Tests/DuplexPipeMarshalingTests.cs

```diff
@@ -786,6 +832,39 @@ public Task AcceptPipeAndChatLater(bool writeOnOdd, IDuplexPipe pipe, Cancellati
         public Task RejectCall(IDuplexPipe pipe) => Task.FromException(new InvalidOperationException("Expected test exception."));
 
         public object ReturnPipeAsObject() => FullDuplexStream.CreatePipePair().Item1;
+
+        public Stream ServerMethodThatReturnsTwoWayStream()
+        {
+            var streamPair = FullDuplexStream.CreatePair();
```

Comment:
These two server methods aren't demonstrating 2-way communication with the client. I suggest you replace this with a `MemoryStream` which you simply initialize and set Position to 0 before returning it. That would resemble something closer to the typical use case of opening a `FileStream` and returning it.

Demonstrating a two way stream (and/or a stream that is one-way but one the client uses to write to instead of read from) would make for another interesting test. Not that it would exercise unique code but it demonstrates a use case that is possible.

Reply:
And speaking of documenting interesting use cases, please update the `oob_streams.md` doc to explain that `Stream` can be returned (I think the doc says it cannot be) and also demonstrates some uses for it.